endpoints/openrtb2/auction.go

```diff
@@ -314,6 +320,68 @@ func parseTimeout(requestJson []byte, defaultTimeout time.Duration) time.Duratio
 	return defaultTimeout
 }
 
+// mergeBidderParams merges bidder parameters passed at req.ext level with imp[].ext level.
+// Preference is given to parameters at imp[].ext level over req.ext level.
+// Parameters at req.ext level are propagated to adapters as is without any validation.
+func mergeBidderParams(req *openrtb_ext.RequestWrapper) error {
+	var reqBidderParams map[string]json.RawMessage
+
+	reqBidderParams, err := adapters.ExtractBidderParams(req.BidRequest)
+	if err != nil {
+		return err
+	}
+
+	impCpy := make([]openrtb2.Imp, 0)
+	for _, imp := range req.BidRequest.Imp {
+		updatedImp := imp
+		var impExt map[string]map[string]interface{}
+		if len(imp.Ext) == 0 {
+			impCpy = append(impCpy, updatedImp)
+			continue
+		}
+
+		err := json.Unmarshal(imp.Ext, &impExt)
+		if err != nil {
+			return fmt.Errorf("error decoding imp[].ext : %s", err.Error())
+		}
+
+		for bidder, impParams := range impExt {
```

Comment:
Currently the preferred location for bidder parameters is `imp[].ext.prebid. bidder.BIDDER`, rather than `imp[].ext.BIDDER`. We have code further down the chain to preserve backwards compatibility, which makes merging at this point in the code more difficult.

 However since this is a new feature, I would be happy with merging only working with bidder parameters in the newer preferred location, rather than the legacy location as you have it now. This should just be simple change digging a little deeper to find the map of bidders to bidder parameters in `imp[].ext.prebid.bidder` rather than `imp[].ext`.

Reply:
+1. There be dragons here with the two locations available. We should also be careful as this is the 3rd implementation of code which searches for bidder exts. The code in `splitImps` is the newest iteration. Perhaps parts can be extracted to a method and used here? Or as @hhhjort said, just make this available for the newer `imp[].ext.prebid.bidder` location. I'm not sure if that would be too confusing for publishers.

Reply:
Thanks @hhhjort & @SyntaxNode for review. I have updated pull request with code review comments. 
Currently I am merging request level bidder params at both new (imp[].ext.prebid.bidder) level & old (imp[].ext) level.
Let me know if I should remove support for old level from pull request.